### PR TITLE
Feature/js tests foundation

### PR DESCRIPTION
**Pull request by xahgmah**
https://trello.com/c/WCmHLZ5I/155-2-js-tests-foundation

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,16 +19,20 @@ cache:
...
   - make tools
+  - make deps-test
+  # Prepare for js tests
+  - make prepare-for-js
 
 script:
   - make quality-py
   - make quality-js
   - make test-py
+  - make test-js
```
Comment (z4y4ts) [inline .travis.yml]: Now you can just leave `make test` since it depends on both test-py & test-js

File: Makefile [hunk 2]
```diff
@@ -22,13 +25,22 @@ quality-js:
...
 deps-test:
 	pip install -r test_requirements.txt
+	bower install
 
 tools:
-	npm install "eslint@^2.12.0" eslint-config-edx "eslint-plugin-dollar-sign@0.0.5" "eslint-plugin-import@^1.9.2"
+	npm install bower "eslint@^2.12.0" eslint-config-edx "eslint-plugin-dollar-sign@0.0.5" "eslint-plugin-import@^1.9.2"
+
+send_coveralls:
+	coveralls-lcov -v -n video_xblock/static/coverage/PhantomJS\ 2.1.1\ \(Linux\ 0.0.0\)/lcov.info > coverage.json
```
Comment (z4y4ts) [inline Makefile]: Is it possible to enquote `.info` file name to avoid backslashes? I.e. `".../PhantomJS 2.1.1 .../lcov.info"`

File: Makefile [hunk 3]
```diff
@@ -22,13 +25,22 @@ quality-js:
...
 
 deps-test:
 	pip install -r test_requirements.txt
+	bower install
 
 tools:
-	npm install "eslint@^2.12.0" eslint-config-edx "eslint-plugin-dollar-sign@0.0.5" "eslint-plugin-import@^1.9.2"
+	npm install bower "eslint@^2.12.0" eslint-config-edx "eslint-plugin-dollar-sign@0.0.5" "eslint-plugin-import@^1.9.2"
+
+send_coveralls:
```
Comment (z4y4ts) [inline Makefile]: Call it `coveralls` for brevity.

File: Makefile [hunk 4]
```diff
@@ -22,13 +25,22 @@ quality-js:
...
+	coveralls-lcov -v -n video_xblock/static/coverage/PhantomJS\ 2.1.1\ \(Linux\ 0.0.0\)/lcov.info > coverage.json
+	coveralls --merge=coverage.json
+
+prepare-for-js:
+	export DISPLAY=:99.0
+	sh -e /etc/init.d/xvfb start
+	npm install jasmine-core karma karma-jasmine karma-phantomjs-launcher karma-coverage karma-coveralls karma-chrome-launcher
```
Comment (z4y4ts) [inline Makefile]: Move this line to `deps-test`

File: Makefile [hunk 5]
```diff
@@ -22,13 +25,22 @@ quality-js:
...
-	npm install "eslint@^2.12.0" eslint-config-edx "eslint-plugin-dollar-sign@0.0.5" "eslint-plugin-import@^1.9.2"
+	npm install bower "eslint@^2.12.0" eslint-config-edx "eslint-plugin-dollar-sign@0.0.5" "eslint-plugin-import@^1.9.2"
+
+send_coveralls:
+	coveralls-lcov -v -n video_xblock/static/coverage/PhantomJS\ 2.1.1\ \(Linux\ 0.0.0\)/lcov.info > coverage.json
+	coveralls --merge=coverage.json
+
+prepare-for-js:
+	export DISPLAY=:99.0
+	sh -e /etc/init.d/xvfb start
```
Comment (z4y4ts) [inline Makefile]: Move these two lines to `test-js`
Probably you also need to add `xvfb stop` after tests have ran

File: video_xblock/backends/base.py [hunk 6]
```diff
@@ -111,55 +111,46 @@ def get_frag(self, **context):
...
-        ))
+
+        js_files = [
+            'static/bower_components/video.js/dist/video.min.js',
+            'static/bower_components/videojs-contextmenu/dist/videojs-contextmenu.min.js',
+            'static/bower_components/videojs-contextmenu-ui/dist/videojs-contextmenu-ui.min.js',
+            'static/js/video-speed.js',
+            'static/js/player_state.js',
+            'static/js/videojs-speed-handler.js'
+
```
Comment (z4y4ts) [inline video_xblock/backends/base.py]: Empty line

File: video_xblock/backends/base.py [hunk 7]
```diff
@@ -111,55 +111,46 @@ def get_frag(self, **context):
...
-        frag.add_javascript(self.resource_string(
-            'static/js/video-speed.js'
-        ))
+        css_files = [
+            'static/bower_components/video.js/dist/video-js.min.css',
+            'static/css/videojs.css',
+            'static/css/videojs-contextmenu-ui.css',
+        ]
+        for css_file in css_files:
+            frag.add_css(self.resource_string(css_file))
```
Comment (z4y4ts) [inline video_xblock/backends/base.py]: 👍

File: video_xblock/static/js/context.js [hunk 8]
```diff
@@ -0,0 +1,6 @@
+/**
+ * Get python context and fill in js global variables with it
+ */
+window.video_player_id = '{{ video_player_id }}';
+window.player_state_obj = JSON.parse('{{ player_state }}');
```
Comment (z4y4ts) [inline video_xblock/static/js/context.js]: Use camelCase.

File: video_xblock/static/js/spec/player_state_spec.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    };
+    window.video_player_id = playerId;
+    beforeEach(function() {
+        var video = document.createElement('video');
+        video.id = playerId;
+        video.className = 'video-js vjs-default-skin';
+        document.body.appendChild(video);
+    });
+    it('return download transcript url', function() {
+        expect(getDownloadTranscriptUrl(player)).toBe(transcripts['en'].url); // eslint-disable-line
```
Comment (z4y4ts) [inline video_xblock/static/js/spec/player_state_spec.js]: Why eslint-disable?
It's a bad practice to mute linter.

File: video_xblock/static/video_xblock_karma.conf.js [hunk 10]
```diff
@@ -0,0 +1,85 @@
+// Karma configuration
+// Generated on Wed Jan 25 2017 15:01:46 GMT+0000 (UTC)
+
+module.exports = function(config) {
+  config.set({
```
Comment (z4y4ts) [inline video_xblock/static/video_xblock_karma.conf.js]: Please use 4 spaces for indentation.

File: video_xblock/static/js/spec/player_state_spec.js (hunk 9 above)
Comment (xahgmah) [inline video_xblock/static/js/spec/player_state_spec.js]:
```
15:16  error  'getDownloadTranscriptUrl' is not defined  no-undef
15:55  error  'transcripts' is not defined               no-undef
```

**Pull request merged**